src/storage/mapping/ExtensionBasedMapper.ts

```diff
@@ -64,10 +64,12 @@ export class ExtensionBasedMapper extends BaseFileIdentifierMapper {
     // If the extension of the identifier matches a different content-type than the one that is given,
     // we need to add a new extension to match the correct type.
     } else if (contentType !== await this.getContentTypeFromPath(filePath)) {
-      const extension: string = mime.extension(contentType) || this.customExtensions[contentType];
+      let extension: string = mime.extension(contentType) || this.customExtensions[contentType];
       if (!extension) {
-        this.logger.warn(`No extension found for ${contentType}`);
-        throw new NotImplementedHttpError(`Unsupported content type ${contentType}`);
+        // When no extension is found for the provided content-type, fall back to application/octet-stream.
+        extension = mime.extension(APPLICATION_OCTET_STREAM) as string;
+        // Signal the fall back by setting the content-type to application/octet-stream in the output link.
+        contentType = APPLICATION_OCTET_STREAM;
```

Comment:
I don't fully remember the specifics of how these classes work and why certain values are set (hence the question), so will require some investigation on how to make this work. Some extra exceptions in the code might be needed to take care of the cases where `contentType` is undefined. If it seems that there is no decent solution that doesn't fully break everything, or it would require a full rewrite of the classes we can still go for the octet stream solution. I would just prefer the `undefined` solution since that is semantically more correct (it indicates that the mapper can't determine the content-type).

Reply:
> I would just prefer the undefined solution since that is semantically more correct (it indicates that the mapper can't determine the content-type).

I fully agree! I will see if I can make it work!

Reply:
A simple solution is to check for the unknown extension that was set in `ExtensionBasedMapper` before calling `getContentTypeFromUrl` in `BaseFileIdentifierMapper`.